src/Parser/DeathKnight/Unholy/Modules/Items/ColdHeartEfficiency.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import ITEMS from 'common/ITEMS';
+import { formatPercentage } from 'common/format';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+class ColdHeartEfficiency extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  on_initialized() {
+    this.active = this.combatants.selected.hasChest(ITEMS.COLD_HEART.id);
+  }
+
+  totalColdHeartCasts = 0;
+  correctColdHeartCasts = 0;
+  buffColdHeart = 0;
+
+  on_byPlayer_applybuff(event){
+    this.addStack(event);
+  }
+
+  on_byPlayer_applybuffstack(event){
+    this.addStack(event);
+  }
+
+  addStack(event){
+    const spellID = event.ability.guid;
+    if(spellID === SPELLS.COLD_HEART_BUFF.id){
+      this.buffColdHeart+=1;
+	if(this.buffColdHeart > 20) {
```

Comment:
Is it possible for buffColdHeart to ever go above 20?  Once its at 20 stacks we shouldn't be seeing any more events for it until there's a cast

Reply:
I was afraid the program would count over 20. I tried removing that bit and it broke the program, showed 0%.

Reply:
Why is addStack even it's own logic, you could just put it all in ``on_byPlayer_applybuffstack`` ? 
If it can't go above 20 however, it shouldn't trigger a ``on_byPlayer_applybuffstack`` event once it's at 20 stacks. You can somewhat see this logic in marksmanship legendary WarbeltOfTheSentinelArmy module for reference.

Reply:
It took the logic from another module. I could try to do it a different way but I don't really see a reason to do so if the current code works.